lib/swap/NodeSwitch.ts

```diff
@@ -35,9 +43,37 @@ class NodeSwitch {
     private readonly logger: Logger,
     cfg?: NodeSwitchConfig,
   ) {
-    this.clnAmountThreshold =
-      cfg?.clnAmountThreshold || NodeSwitch.defaultClnAmountThreshold;
-    this.logger.info(`CLN invoice threshold: ${this.clnAmountThreshold} sat`);
+    if (cfg?.clnAmountThreshold !== undefined) {
+      if (typeof cfg.clnAmountThreshold === 'number') {
+        this.clnAmountThreshold = {
+          [SwapType.Submarine]: cfg.clnAmountThreshold,
+          [SwapType.ReverseSubmarine]: cfg.clnAmountThreshold,
+        };
+      } else {
+        this.clnAmountThreshold = {
+          [SwapType.Submarine]:
+            cfg.clnAmountThreshold.submarine ||
+            NodeSwitch.defaultClnAmountThreshold,
+          [SwapType.ReverseSubmarine]:
+            cfg.clnAmountThreshold.reverse ||
+            NodeSwitch.defaultClnAmountThreshold,
+        };
+      }
+    } else {
+      this.clnAmountThreshold = {
+        [SwapType.Submarine]: NodeSwitch.defaultClnAmountThreshold,
+        [SwapType.ReverseSubmarine]: NodeSwitch.defaultClnAmountThreshold,
+      };
+    }
```

Comment:
```suggestion
    }
    this.clnAmountThreshold[SwapType.Submarine] ??= NodeSwitch.defaultClnAmountThreshold
    this.clnAmountThreshold[SwapType.Reverse] ??= NodeSwitch.defaultClnAmountThreshold
```
One can also remove the `||` operators in above this way.

Reply:
Can't because `this.clnAmountThreshold` itself is undefined before this statement